lib/csv.rb

```diff
@@ -2260,25 +2260,56 @@ def header_convert(name = nil, &converter)
 
   include Enumerable
 
+  # :call-seq:
+  #   csv.each -> self
   #
-  # Yields each row of the data source in turn.
+  # Calls the block with each successive row.
+  # The data source must be opened for reading.
   #
-  # Support for Enumerable.
+  # Without headers:
+  #   string = "foo,0\nbar,1\nbaz,2\n"
+  #   csv = CSV.new(string)
+  #   csv.each do |row|
+  #     p row
+  #   end
+  # Output:
+  #   ["foo", "0"]
+  #   ["bar", "1"]
+  #   ["baz", "2"]
   #
-  # The data source must be open for reading.
+  # With headers:
+  #   string = "Name,Value\nfoo,0\nbar,1\nbaz,2\n"
+  #   csv = CSV.new(string, headers: true)
+  #   csv.each do |row|
+  #     p row
+  #   end
+  # Output:
+  #   <CSV::Row "Name":"foo" "Value":"0">
+  #   <CSV::Row "Name":"bar" "Value":"1">
+  #   <CSV::Row "Name":"baz" "Value":"2">
+  #
+  # ---
   #
+  # Raises an exception if the source is not opened for reading:
+  #   string = "foo,0\nbar,1\nbaz,2\n"
+  #   csv = CSV.new(string)
+  #   csv.close
+  #   # Raises IOError (not opened for reading)
+  #   csv.each do |row|
+  #     p row
+  #   end
   def each(&block)
     parser_enumerator.each(&block)
   end
 
   # :call-seq:
-  #     read
+  #   csv.read -> array or csv_table
```

Comment:
How about splitting each pattern?

```suggestion
  #   csv.read -> array
  #   csv.read -> csv_table (CSV::Table is better?)
```

Reply:
> What I have here is what I see in the doc for such as Hash and Array.  People have been careful to use something (csv_table) that is English, not code.  I once wrote foo.to_h -> Hash, and had someone point out that it's somewhat ambiguous:  class Hash or instance of Hash?

Reply:
I see.